model/task_lifecycle.go

```diff
@@ -798,6 +803,26 @@ func MarkEnd(ctx context.Context, settings *evergreen.Settings, t *task.Task, ca
 	return nil
 }
 
+func handleRetryableTask(detail *apimodels.TaskEndDetail, t *task.Task) error {
+	if detail == nil || !detail.Retryable {
+		return nil
+	}
+	taskToReset := t
+	if t.IsPartOfDisplay() {
+		dt, err := t.GetDisplayTask()
+		if err != nil {
+			return errors.Wrap(err, "getting display task")
+		}
+		taskToReset = dt
+	}
+	if taskToReset.NumAutomaticResets < evergreen.MaxAutomaticRestarts {
```

Comment:
Would be nice if there was a way to send an HTTP error back if they try requesting more than the maximum auto-restarts.